features/layer_lock.c

```diff
@@ -21,6 +21,37 @@
 // The current lock state. The kth bit is on if layer k is locked.
 static layer_state_t locked_layers = 0;
 
+#if LAYER_LOCK_IDLE_TIMEOUT > 0
+#if LAYER_LOCK_IDLE_TIMEOUT < 100 || LAYER_LOCK_IDLE_TIMEOUT > 30000
+//Constrain timeout to a sensible range. With the 16-bit timer, the longest
+//representable timeout is 32768 ms, rounded here to 30000 ms = half a minute.
+#error "layer_lock: LAYER_LOCK_IDLE_TIMEOUT must be between 100 and 30000 ms"
+#endif
+
+//Layer Lock timer to disable layer lock after X seconds inactivity
+
+static uint16_t layer_lock_timer = 0;
+
+    void layer_lock_timer_task(void) {
+        const uint8_t layer = get_highest_layer(layer_state);
+        if (is_layer_locked(layer) && timer_elapsed(layer_lock_timer) > LAYER_LOCK_IDLE_TIMEOUT) {
```

Comment:
A (perhaps excessive) feature of this library is that each layer may be locked independently, and it is possible that multiple layers are locked at the same time. In that edge case, this logic would unlock the highest layer, then wait the timeout again before unlocking the next highest layer, and so on. It's probably better more predictable behavior to unlock all layers at once. How about add this helper:

```c
void layer_lock_all_off(void) {
  layer_and(~locked_layers);
  locked_layers = 0;
}
```

Then change the task function to do

```c
if (locked_layers && timer_elapsed(layer_lock_timer) > LAYER_LOCK_IDLE_TIMEOUT) {
  layer_lock_all_off();
}
```

Reply:
I did not know it could lock multiple layers, definitely need some protection against that so I'll make the suggested changes and retest.